bin/live/pycbc_live_single_trigger_fits

```diff
@@ -198,110 +191,219 @@ for filename in files:
-                        file_array.data['mass1'], file_array.data['mass2'],
-                        file_array.data['spin1z'], file_array.data['spin2z'])
-                    file_array.data[cparam] = file_vals
-                #Apply cuts to array for this file
-                cineq_fn = inequality_string_to_function(cineq)
-                idx_keep = np.nonzero(cineq_fn(file_array.data[cparam],
-                                               cthresh))[0]
-                # if nothing remains after cuts, skip file
-                if not len(idx_keep):
-                    skipping_file = True
-                    break
-                file_array = file_array.select(idx_keep)
-            if args.cluster:
-                if args.cluster=='new_snr':
-                    #Calculating new_snr for clustering
-                    file_array.data['new_snr'] = ranking.newsnr(
-                                                     file_array.data['snr'],
-                                                     file_array.data['chisq'])
-                max_idx = file_array.data[args.cluster].argmax()
-                file_array = file_array.select([max_idx])
-            if skipping_file: continue
-            events[ifo] = events[ifo] + file_array
 
-logging.info("All events processed")
+            # Read trigger value datasets from file
+            triggers[ifo] = {k: fin[ifo][k][:] for k in fin[ifo].keys()
+                             if k not in ('loudest', 'stat', 'gates', 'psd')
+                             and fin[ifo][k].size == n_triggers}
+            # The n_triggers filter should be enough, but the 'not in'
+            # list is for edge cases where the number of gates/loudest etc is
+            # the same as the number of triggers
 
+            # The stored chisq is actually reduced chisq, so we need to
+            # alter the chisq_dof dataset so we can use the standard conversions.
+            # chisq_dof of 1.5 gives the right number (2 * 1.5 - 2 = 1)
+            triggers[ifo]['chisq_dof'] = \
+                1.5 * np.ones_like(triggers[ifo]['snr'])
 
-# calculate new_snr and cut if requested
-if cutting_newsnr or args.fit_param == 'new_snr' or 'new_snr' not in events[ifo].data:
-    logging.info("Calculating new_snr")
-    for ifo in args.ifos:
-        events[ifo].data['new_snr'] = ranking.newsnr(events[ifo].data['snr'],
-                                                     events[ifo].data['chisq'])
-#Note: according to Alex, the stored chisq here is already the reduced chisq
 
-if cutting_newsnr:
-    logging.info("Keeping events with new_snr %s %.3f" % (nsineq[0], nsthresh))
-    ineq_fn = inequality_string_to_function(nsineq)
     for ifo in args.ifos:
-        idx_keep = np.nonzero(ineq_fn(events[ifo].data['new_snr'], nsthresh))[0]
-        if len(idx_keep):
-            events[ifo] = events[ifo].select(idx_keep)
+        # Continue if not triggers found in file
+        if ifo not in triggers:
+            continue
```

Comment:
If you simply iterate over `triggers` instead of over `args.ifos`, you can skip this check, right?